x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json

```diff
@@ -0,0 +1,974 @@
+    {
+      "rename": {
+        "field": "checkpoint.client_inbound_interface",
+        "target_field": "observer.egress.interface.name",
+        "ignore_missing": true,
+        "if": "ctx.network?.direction == 'outbound'"
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.type",
+        "target_field": "observer.type",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.origin",
+        "target_field": "observer.name",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.origin_ip",
+        "target_field": "observer.ip",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.endpoint_ip",
+        "target_field": "observer.ip",
+        "ignore_missing": true,
+        "if": "ctx.observer?.ip == null"
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.outzone",
+        "target_field": "observer.egress.zone",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.inzone",
+        "target_field": "observer.ingress.zone",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.security_outzone",
+        "target_field": "observer.egress.zone",
+        "ignore_missing": true,
+        "if": "ctx.observer?.egress?.zone == null"
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.security_inzone",
+        "target_field": "observer.ingress.zone",
+        "ignore_missing": true,
+        "if": "ctx.observer?.ingress?.zone == null"
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.update_version",
+        "target_field": "observer.version",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.process_md5",
+        "target_field": "process.hash.md5",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.process_name",
+        "target_field": "process.name",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.parent_process_md5",
+        "target_field": "process.parent.hash.md5",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.parent_process_name",
+        "target_field": "process.parent.name",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.matched_category",
+        "target_field": "rule.category",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.categories",
+        "target_field": "rule.category",
+        "ignore_missing": true,
+        "if": "ctx.rule?.category == null"
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.malware_action",
```

Comment:
we could probably use the presense of "checkpoint.malware_action" to add "malware" to event.category before you rename it.